handbook/product/design/design-and-interaction-guidelines.md

```diff
@@ -0,0 +1,107 @@
+# Design and interaction guidelines
+
+Sourcegraph’s search experience is the application’s primary interaction. As such, the design is tailored to the search use case by prioritizing code over other elements.
+
+With the addition of campaigns, repository groups, and code insights, the product is quickly evolving. As such, its design must evolve to meet the experience requirements of these new features. 
+
+This document will describe the current design and interaction models and describe the principles we will utilize to inform our design and interaction choices moving forward.
+
+## Two design styles
+Sourcegraph utilizes two design and interaction styles which are tailored to the content they display:
+
+Search - An IDE like, full screen, high density, low contrast ui (for elements other than code) which centers on the display of code
+Web application - a more traditional web application style with larger margins and padding, increased contrast and standard web page design experience requirements
+
+## Code pages design
+
+In Sourcegraph’s search and file experiences, code is the most important element. The design of these pages reflects that priority. Contrast is balanced so that code stands out above all else. Panels clearly segment content and tools from the code. Pages are full width, and margins and padding are slim to increase the density of information which helps us display as much code on the screen as possible. 
+
+The interaction model of the search experience is also unique. Clicking text on the page will almost always result in running a query or viewing a file. In a normal web application, we would clearly highlight these interactive elements so that the user understands where the actions are. If we were to do so on search pages, these elements would quickly outweigh and overwhelm the code. 
+
+### Examples:
+
+* [Search results](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:.*/go+auth&patternType=literal)
+* [File views](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/package.json)
+
+<img src='./margins-example.png' />
+
+_Small margins and reduced padding help to minimize scanning and scrolling. This allows developers to see code faster… and fit more of it on the screen._
+
+
+### Icons and labels
```

Comment:
I would consider changing the order here:

1. Code page design
2. Web app page design
3. Icons and labels - both for code page and web page -> it will be easier to compare those approached without jumping between the sections


@rrhyne @quinnkeast wdyt?

Reply:
Fine with me, would you mind making that change?

Reply:
I went to address that and realized I didn't understand what you meant. The icons and labels are sub-sections of each main topic (code pages and web pages) as they are treated in different ways for each. 

So in that sense, I don't think re-ordering makes sense. If Quinn does, we can address this in a separate pull request.

Reply:
@quinnkeast wdyt?